onnxscript/rewriter/llama_rule_sets.py

```diff
@@ -2,13 +2,73 @@
 # Licensed under the MIT License.
 from __future__ import annotations
 
+import numpy as np
+import onnx.numpy_helper
+
 import onnxscript.ir as ir
 import onnxscript.rewriter.no_op as no_op
 import onnxscript.rewriter.pattern as orp
 
 op = orp.onnxop
 
 
+class CastIdentity(orp.RewriteRuleAsClass):
+    """Replaces ``Cast(., to=to)`` by ``Identity`` if possible."""
+
+    @classmethod
+    def pattern(cls, op, x, to: int):
+        return op.Cast(x, to=to)
+
+    @classmethod
+    def rewrite(cls, op, x: ir.Value, to: int):
+        return op.Identity(x)
+
+    @classmethod
+    def check(cls, context, x, to: int) -> bool:
+        return x.dtype == to.value
+
+
+class CastCast(orp.RewriteRuleAsClass):
+    """Replaces ``Cast(Cast(X, ...), to=to)`` by ``Cast(X, to=to)``."""
+
+    @classmethod
+    def pattern(cls, op, x, to: int, to0: int):
+        return op.Cast(op.Cast(x, to=to0), to=to)
+
+    @classmethod
+    def rewrite(cls, op, x: ir.Value, to: int, to0: int):
+        return op.Cast(x, to=to)
+
+
+class ExpandIdentity(orp.RewriteRuleAsClass):
+    """Replaces ``Expand(., shape)`` by ``Identity`` if possible."""
+
+    @classmethod
+    def pattern(cls, op, x, shape):
+        return op.Expand(x, shape)
+
+    @classmethod
+    def rewrite(cls, op, x: ir.Value, shape: ir.Value):
+        return op.Identity(x)
+
+    @classmethod
+    def check(cls, context, x, shape) -> bool:
+        shape_x = x.shape
```

Comment:
I think you need an extra check to check that `shape` is a constant?

Reply:
We might have some to-do task relating to this, let me search

Reply:
It would be better. I just hope it fails right now.

Reply:
I see some example [here](https://github.com/microsoft/onnxscript/blob/47fb031b98da291d005a71a782f29f72514db806/onnxscript/rewriter/pattern_test.py#L261) ... so, two points: 
(i) Checking required is simpy checking `shape.const_value is not None`
(ii) The TODO item is to integrate constant-propagation into the IR/rewriter. For now, the workaround used is as shown in above example to use `_ir_utils.propagate_const_value`.

Reply:
Let's check it is a constant. We may do more if needed later.